pyface/tests/test_color.py

```diff
@@ -0,0 +1,246 @@
+            with self.subTest(int=value):
+                result = channels_to_ints(
+                    ints_to_channels(
+                        [value],
+                        maximum=65535,
+                    ),
+                    maximum=65535,
+                )
+                self.assertEqual(result, (value,))
+
+
+
+class TestColor(UnittestTools, TestCase):
+
+    def test_init(self):
+        color = Color()
+        self.assertEqual(color.rgba, (1.0, 1.0, 1.0, 1.0))
+
+    def test_init_rgba(self):
+        color = Color(rgba=(0.4, 0.2, 0.6, 0.8))
+        self.assertEqual(color.rgba, (0.4, 0.2, 0.6, 0.8))
+
+    def test_init_rgb(self):
+        color = Color(rgb=(0.4, 0.2, 0.6))
+        self.assertEqual(color.rgba, (0.4, 0.2, 0.6, 1.0))
+
+    def test_init_r_g_b_a(self):
+        color = Color(red=0.4, green=0.2, blue=0.6, alpha=0.8)
+        self.assertEqual(color.rgba, (0.4, 0.2, 0.6, 0.8))
+
+    def test_init_r_g_b(self):
+        color = Color(red=0.4, green=0.2, blue=0.6)
+        self.assertEqual(color.rgba, (0.4, 0.2, 0.6, 1.0))
+
+    def test_init_hsva(self):
+        color = Color(hsva=(0.4, 0.2, 0.6, 0.8))
+        self.assertEqual(color.rgba, (0.48, 0.6, 0.528, 0.8))
+
+    def test_init_hsv(self):
+        color = Color(hsv=(0.4, 0.2, 0.6))
+        self.assertEqual(color.rgba, (0.48, 0.6, 0.528, 1.0))
+
+    def test_init_hlsa(self):
+        color = Color(hlsa=(0.4, 0.2, 0.6, 0.8))
+        self.assertEqual(
+            color.rgba,
+            (0.07999999999999996, 0.32000000000000006, 0.17600000000000007, 0.8)  # noqa: E501
+        )
+
+    def test_init_hls(self):
+        color = Color(hls=(0.4, 0.2, 0.6))
+        self.assertEqual(
+            color.rgba,
+            (0.07999999999999996, 0.32000000000000006, 0.17600000000000007, 1.0)  # noqa: E501
+        )
+
+    def test_toolkit_round_trip(self):
+        color = Color(rgba=(0.4, 0.2, 0.6, 0.8))
+        toolkit_color = color.to_toolkit()
+        result = Color.from_toolkit(toolkit_color)
+        self.assertEqual(result.rgba, (0.4, 0.2, 0.6, 0.8))
+
+    def test_hex(self):
+        color = Color(rgba=(0.4, 0.2, 0.6, 0.8))
+        hex_value = color.hex()
+        self.assertEqual(hex_value, "#663399CC")
+
+    def test_hex_black(self):
+        color = Color(rgba=(0.0, 0.0, 0.0, 1.0))
+        hex_value = color.hex()
+        self.assertEqual(hex_value, "#000000FF")
+
+    def test_eq(self):
+        color_1 = Color(rgba=(0.4, 0.2, 0.6, 0.8))
+        color_2 = Color(rgba=(0.4, 0.2, 0.6, 0.8))
+        self.assertTrue(color_1 == color_2)
+        self.assertFalse(color_1 != color_2)
+
+    def test_eq_not_eequal(self):
+        color_1 = Color(rgba=(0.4, 0.2, 0.6, 0.8))
+        color_2 = Color(rgba=(0.4, 0.4, 0.6, 0.8))
+        self.assertTrue(color_1 != color_2)
+        self.assertFalse(color_1 == color_2)
+
+    def test_eq_other(self):
+        color = Color(rgba=(0.4, 0.2, 0.6, 0.8))
+        self.assertFalse(color == 1)
```

Comment:
This is a bit confusing. Should this comparison return True/False? I was under the impression that this would return an Exception.

Reply:
Equality comparisons should almost always return either True or False for almost any pair of Python objects - the the comparison doesn't make sense, then they aren't equal.  There are exceptions (numpy arrays!), but those exceptions make code much more difficult to write when they occur.

Reply:
> Objects of different types, except different numeric types, never compare equal.

from the Python docs - https://docs.python.org/3/library/stdtypes.html#comparisons. I didn't recall this. resolving.